### PR TITLE
Density docs

### PR DESCRIPTION
**Pull request by marxwillia**
Added warnings and updated documentation relating to Issue #908. Warnings alert the user when the boundary velocities specified in the config file are ignored.  Explanation of this behavior is added to the custom density documentation.

### CODE REVIEW

File: tardis/model/base.py [hunk 1]
```diff
@@ -220,20 +220,23 @@ def no_of_raw_shells(self):
     def v_boundary_inner(self):
         if self._v_boundary_inner is None:
             return self.raw_velocity[0]
+        if self._v_boundary_inner < 0:
```
Comment (wkerzendorf) [inline tardis/model/base.py]: maybe you want to specify `0 * u.km/u.s`

File: tardis/model/base.py [hunk 2]
```diff
@@ -220,20 +220,23 @@ def no_of_raw_shells(self):
...
-            value = u.Quantity(value, self.v_boundary_inner.unit)
-            if value > self.v_boundary_outer:
-                raise ValueError('v_boundary_inner must not be higher than '
-                                 'v_boundary_outer.')
-            if value > self.raw_velocity[-1]:
-                raise ValueError('v_boundary_inner is outside of '
-                                 'the model range.')
-            if value <= self.raw_velocity[0]:
-                value = None
+            if value > 0:
```
Comment (wkerzendorf) [inline tardis/model/base.py]: again `0 * u.km / u.s`

File: tardis/model/base.py [hunk 3]
```diff
@@ -242,20 +245,23 @@ def v_boundary_inner(self, value):
     def v_boundary_outer(self):
         if self._v_boundary_outer is None:
             return self.raw_velocity[-1]
+        if self._v_boundary_outer < 0:
```
Comment (wkerzendorf) [inline tardis/model/base.py]: again `0 * u.km/u.s`

File: tardis/model/base.py [hunk 4]
```diff
@@ -220,20 +220,23 @@ def no_of_raw_shells(self):
...
+            if value > 0:
+                value = u.Quantity(value, self.v_boundary_inner.unit)
+                if value > self.v_boundary_outer:
+                    raise ValueError('v_boundary_inner must not be higher than '
+                                     'v_boundary_outer.')
+                if value > self.raw_velocity[-1]:
+                    raise ValueError('v_boundary_inner is outside of '
+                                     'the model range.')
+                if value < self.raw_velocity[0]:
+                    raise ValueError('v_boundary_inner is outside of the model range.')
```
Comment (wkerzendorf) [inline tardis/model/base.py]: is lower than the lowest shell in the model

File: docs/examples/densitycust.rst [hunk 5]
```diff
@@ -42,6 +42,13 @@ The values in the example here define a density profile that is dropping off wit
...
 
+Inner Boundary
+==============
+
+The first velocity-density pair in a custom density file (given by index 0) specifies the velocity of the inner boundary approximation to the photosphere. The density associated with this velocity is the density of the optically thick inner region, which does not affect TARDIS spectra. Therefore, the first density (5.4869692e-10 in the example above) can be replaced by a placeholder value. The user can choose to both specify a custom density fileAND specify v_inner_boundary and v_outer_boundary in the configuration YAML file for a TARDIS run. In this case, v_inner_boundary is ignored if it is less than the smallest velocity listed in the custom density file. Likewise, v_outer_boundary is ignored if it is larger than the largest velocity in the custom density file. If either of the YAML specified boundary velocities fall within the velocity range specified in the custom density file, then the boundary velocity is set equal to the number in the configuration YAML file. This has the effect of splitting a cell in the custom density file into two parts, a region within the boundary and a region outside the boundary.
```
[the remark this answers is not in the document]
Comment (marxwillia) [reply docs/examples/densitycust.rst]: We've decided not to implement this functionality.  The user must change their model if they want to specify a v_inner_boundary lower than the lowest shell velocity, and similarly for v_outer_boundary.

**Comment by wkerzendorf** [comment]
Looks very good - what happened to the notebook?

File: docs/yml_files/densitycust/test_config.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
+# Example YAML configuration for TARDIS
+tardis_config_version: v1.0
+
+supernova:
+  luminosity_requested: 8.770 log_lsun
+  time_explosion: 16 day
+  distance: 8.32 Mpc
+
+# standard atomic data base; get it from the tardis-refdata repository
+atom_data: /home/mew488/Research/TARDIS/tardis-refdata/atom_data/kurucz_cd23_chianti_H_He.h5
```
Comment (wkerzendorf) [inline docs/yml_files/densitycust/test_config.yml]: now you can just write `kurucz_cd23_chianti_H_He.h5`

**Pull request merged**